File: TestEventProducer/src/main/java/com/mrozowski/testeventproducer/serialization/InventoryItemEventMessageDeserializer.java
package com.mrozowski.testeventproducer.serialization;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.mrozowski.testeventproducer.InventoryItemEventMessage;
import org.apache.kafka.common.serialization.Deserializer;

import java.io.IOException;

public class InventoryItemEventMessageDeserializer implements Deserializer<InventoryItemEventMessage> {
  private final ObjectMapper objectMapper = new ObjectMapper();

  @Override
  public InventoryItemEventMessage deserialize(String topic, byte[] data) {
    try {
      return objectMapper.readValue(data, InventoryItemEventMessage.class);
    } catch (IOException e) {
      throw new RuntimeException("Error deserializing InventoryItemEventMessage", e);
    }
  }
}

File: src/main/kotlin/com/mrozowski/ims/adapter/incoming/InventoryItemEventMessageDeserializer.kt
package com.mrozowski.ims.adapter.incoming

import com.fasterxml.jackson.databind.ObjectMapper
import org.apache.kafka.common.serialization.Deserializer
import java.io.IOException

class InventoryItemEventMessageDeserializer : Deserializer<InventoryItemEventMessage> {
    private val objectMapper = ObjectMapper()

    override fun deserialize(topic: String?, data: ByteArray?): InventoryItemEventMessage {
        return try {
            objectMapper.readValue(data, InventoryItemEventMessage::class.java)
        } catch (e: IOException) {
            throw RuntimeException("Error deserializing InventoryItemEventMessage", e)
        }
    }
}